### DISCUSSION TITLE
Is it possible to make react component library using twin.macro?

### DISCUSSION THREAD

**Discussion by jessedelorenzo** [ben-rogerson/twin.macro#592] [Q&A]
I used twin.macro for the last 2 years and am very exciting about it thanks to fast coding and styling.
These days I tried to make a react UI components library as a npm package using twin.macro and import those components in other react components (non twin.macro project).

Is there any possible solution to do this? I really couldn't make it work. It's better if you can add an example boilerplate for the solution.

Thanks.

**Comment by ben-rogerson** [comment]
Hey there,
If you're creating a npm package then you'll need to precompile the UI component library files.
[Take a look at the rollup component library example](https://github.com/ben-rogerson/twin.examples/tree/master/component-library-emotion) which precompiles the `packages/components` files for use in `packages/app`.
